django/publicmapping/publicmapping/templates/index.html

```diff
@@ -134,6 +134,17 @@ <h3>{% trans "About this DistrictBuilder instance" %}</h3>
             <script src="https://connect.facebook.net/en_US/all.js#xfbml=1"></script>
             <fb:like width="90" show_faces="false" action="like" layout="button_count"></fb:like>
           </li>
+
+          <li class="email-share">
+            <a id="email-share-link">
+              <button id="email-share-button"><img id="email-share-img" src="{% static 'images/icon-mail.png' %}"></button>
+            </a>
+          </li>
+
+          <script type="text/javascript">
+            var mailLink = 'mailto:?subject={% trans "Welcome to DistrictBuilder" %}&body=' + window.location.href;
```

Comment:
It looks like we could use `request.build_absolute_uri` to get the full URL and avoid using any JS. I think it could look like this:
```
          <li class="email-share">
            <a id="email-share-link" href="mailto:?subject={% trans "Welcome to DistrictBuilder" %}&body={{ request.build_absolute_uri }}">
              <button id="email-share-button"><img id="email-share-img" src="{% static 'images/icon-mail.png' %}"></button>
            </a>
          </li>
```
Basing that off of: https://simpleisbetterthancomplex.com/tips/2016/07/20/django-tip-7-how-to-get-the-current-url-within-a-django-template.html

Reply:
[This issue](https://www.pivotaltracker.com/n/projects/2140015/stories/157125912) means that we currently can't do that. The options I see are first fixing the issue and then using `request.build_absolute_uri` instead of JS here or just using JS here and fixing the issue later. @ddohler's PR for the Facebook share link should also be affected by this decision. Any thoughts on which path to take? cc: @kshepard

Reply:
I think eventually we do need to fix the `request.build_absolute_uri` issue because for the FB PR it needs to set a `<meta>` tag in order to define the `og:url` property. However, here, it seems pretty easy to accomplish the same thing with Javascript, and easy to change it to use `build_absolute_uri` once it works, so I'm not opposed to leaving a workaround here until we get the fix in place.

Reply:
Sure, I'd forgotten about this issue. I'm fine with the workaround.